custom_components/storj/api.py

```diff
@@ -139,9 +144,28 @@ async def async_delete_backup(self, backup: AgentBackup) -> None:
         if result.returncode != 0:
             raise UplinkError("Unable to delete backup")
 
-    async def async_download_backup(self) -> None:
+    async def async_download_backup(
+        self, backup: AgentBackup, backup_path: Path
+    ) -> AsyncIterator[bytes]:
         """Download a backup to the local system."""
-        _LOGGER.debug("TODO")
+
+        temp_location = str(backup_path.joinpath("temp", suggested_filename(backup)))
+        result = await asyncio.create_subprocess_exec(
+            "uplink",
+            "cp",
+            f"sj://{self.bucket_name}/backups/{suggested_filename(backup)}",
+            temp_location,
+        )
+        await result.communicate()
+        if result.returncode != 0:
+            raise UplinkError("Unable to download temp backup")
+
+        file_obj = await aiofiles.open(temp_location, "rb")
+        iterator = ChunkAsyncStreamIterator(await file_obj.read())
```

Comment:
I found that if I just passed in the file obj and did the reading by chunks in the iterator, I got an error because the file had been closed. This is definitely awkward but I'm not sure what a better alternative is at the moment short of completely re-implementing `uplink` as a python library that is friendlier.